### PR TITLE
new media: upload on save (removed upload button)

### PR DESCRIPTION
**Pull request by didoda**
This provides #207.

No more `Upload` button.
Upload logic is moved to `ModulesComponent` (and tests are update consequently).

On missing `status`, use default 'draft'.
Note: it will be better handled directly on view, through `SchemaHelper` helper (after refactor is done, and https://github.com/bedita/web/pull/206 is merged).

Future feature (in another issue): in "edit media" modify file (upload another file).

### CODE REVIEW

**Comment by didoda** [comment]
Test fail for an unexpected docker behavior. Fix is coming soon (https://github.com/bedita/bedita/pull/1564)

File: src/Controller/Component/ModulesComponent.php [hunk 1]
```diff
@@ -198,4 +199,55 @@ public function objectTypes(?bool $abstract = null) : array
...
     }
+
+    /**
+     * Upload a file and store it in a media stream
+     *
+     * @param BEditaClient $apiClient The api client to perform upload
+     * @param array $requestData The request data from form
+     * @return void
+     */
+    public function upload(BEditaClient $apiClient, array &$requestData) : void
```
Comment (stefanorosanelli) [inline src/Controller/Component/ModulesComponent.php]: API client may be used with `ApiClientProvider::getApiClient()` no need to pass it as argument

File: src/Controller/Component/ModulesComponent.php [hunk 2]
```diff
@@ -198,4 +199,55 @@ public function objectTypes(?bool $abstract = null) : array
...
+        $headers = ['Content-Type' => $requestData['file']['type']];
+        $apiClient = ApiClientProvider::getApiClient();
+        $response = $apiClient->upload($filename, $filepath, $headers);
+
+        // create media from stream
+        $streamId = $response['data']['id'];
+        $type = $requestData['model-type'];
+        $title = $filename;
+        $objectHasStatus = !empty($requestData['status']);
+        $status = $objectHasStatus ? $requestData['status'] : 'draft';
```
[the remark this answers is not in the document]
Comment (didoda) [reply src/Controller/Component/ModulesComponent.php]: Fine. But it means that an API a change is needed. What could we do in here?
I would let it like it is now, and change it after an API proper change

File: src/Template/Pages/Modules/view.twig [hunk 3]
```diff
@@ -32,6 +27,11 @@
             <div class="form-column">
                 {% element 'Form/core_properties' %}
 
+                {# Upload available only for new media objects #}
+                {% if object.id is empty and objectType in ['images', 'files', 'audio', 'videos'] %}
```
[the remark this answers is not in the document]
Comment (didoda) [reply src/Template/Pages/Modules/view.twig]: Not sure. Semantically I'd prefer the order "title", "description", "file".
Note that in the view of an image (with file already uploaded), you see data exactly in this order: "title", "description", <image>.

File: src/Controller/Component/ModulesComponent.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (stefanorosanelli) [reply src/Controller/Component/ModulesComponent.php]: If there's an API bug we should open an issue on `bedita/bedita`
I just tested a `POST /images` with no attributes (without status) and it works, resulting object image has `draft` status... 
What's the problem here?

File: src/Template/Pages/Modules/view.twig (hunk 3 above)
[the remark this answers is not in the document]
Comment (stefanorosanelli) [reply src/Template/Pages/Modules/view.twig]: not sure in this case, anyway it is really `[minor]` and we could keep it like this

**Pull request merged**